commit e98937475a7f03e69a3a3c6a4ee67a8d41f027b5
Author: Andras Mantia <amantia@kde.org>
Date:   Sat Aug 25 11:47:36 2012 +0300

    Now fix for real the mail loss on spam filtering. The previous fix's idea was good, just that unfortunately the default value is "setCacheOnly(true)" set in AgentBasePrivate, so we have to explicitely set the cacheonly to false, otherwise mail content is not fetched from the mail server when passing through a filter that *needs* a content.
    
    FIXED-IN: 4.9.1
    
    BUG: 295484
    (cherry picked from commit c691cb5d9cc7bb337c149e0279e21397f168c314)

diff --git a/mailfilteragent/mailfilteragent.cpp b/mailfilteragent/mailfilteragent.cpp
index 7220d90..75d6f05 100644
--- a/mailfilteragent/mailfilteragent.cpp
+++ b/mailfilteragent/mailfilteragent.cpp
@@ -117,6 +117,7 @@ void MailFilterAgent::initialCollectionFetchingDone( KJob *job )
   Akonadi::CollectionFetchJob *fetchJob = qobject_cast<Akonadi::CollectionFetchJob*>( job );
 
   changeRecorder()->itemFetchScope().setAncestorRetrieval( Akonadi::ItemFetchScope::Parent );
+  changeRecorder()->itemFetchScope().setCacheOnly(false);
   mRequestedPart = m_filterManager->requiredPart();
   if (mRequestedPart == MailCommon::SearchRule::CompleteMessage) {
     changeRecorder()->itemFetchScope().fetchFullPayload();
